### PR TITLE
Add Tcl bindings...

### PR DESCRIPTION
**Pull request by mistachkin**
This pull request adds bindings for use by the Tcl scripting language.  It is designed to provide a very thin wrapper over the low-level C API, e.g. exposing primitive commands that can be used to build more complex and/or specialized layers above it, e.g. TDBC, etc.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -118,6 +118,7 @@ add_subdirectory(protobuf)
 add_subdirectory(schemachange)
 add_subdirectory(sockpool)
 add_subdirectory(sqlite)
+add_subdirectory(tcl)
```
Comment (mponomar) [inline CMakeLists.txt]: We should either document tcldev as a dependency, or make this optional.

File: cmake/FindTcl.cmake [hunk 2]
```diff
@@ -0,0 +1,9 @@
+include(${CMAKE_MODULE_PATH}/pkg_helper.cmake)
+find_pkg_for_comdb2(Tcl
```
Comment (mponomar) [inline cmake/FindTcl.cmake]: I didn't have tcl.h on my system - it passed cmake and failed at compile time - this step should fail if not found (or tcl bindings can be optional and only build if tcl exists).

File: CMakeLists.txt (hunk 1 above)
Comment (mistachkin) [reply CMakeLists.txt]: Good point, I'll need to figure out how to accomplish that with CMake.

File: cmake/FindTcl.cmake (hunk 2 above)
Comment (mistachkin) [reply cmake/FindTcl.cmake]: I'll figure out how to make the necessary changes for CMake; however, I may need your help in testing them as most (other) systems will probably have the Tcl stuff installed?

File: CMakeLists.txt [hunk 3]
```diff
@@ -49,6 +49,11 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL Linux OR ${CMAKE_SYSTEM_NAME} STREQUAL Darwin)
   find_package(Unwind REQUIRED)
 endif()
 
+option(WITH_TCL "Turn ON to compile with Tcl bindings" OFF)
+if(WITH_TCL)
+  find_package(Tcl REQUIRED)
+endif()
+
```
Comment (mistachkin) [inline CMakeLists.txt]: I think this is the "right" way to have an optional feature?

File: CMakeLists.txt [hunk 4]
```diff
@@ -118,6 +123,9 @@ add_subdirectory(protobuf)
 add_subdirectory(schemachange)
 add_subdirectory(sockpool)
 add_subdirectory(sqlite)
+if(WITH_TCL)
+  add_subdirectory(tcl)
+endif()
```
Comment (mistachkin) [inline CMakeLists.txt]: This does seem to work.  It causes the directory to be skipped unless the feature is enabled.

File: cmake/FindTcl.cmake (hunk 2 above)
Comment (mistachkin) [reply cmake/FindTcl.cmake]: This issue should be fixed now.  Tcl support is not included unless enabled.

**Comment by mistachkin** [comment]
From the log files, I do not see a failure of the "yast" test.  It shows as "success"; however, there is a core dump file for the Tcl shell (tclsh).  Not sure where that came from as the logs do not seem to mention Tcl crashing.

**Comment by mistachkin** [comment]
With my most recent check-in on this branch, the "yast" tests now pass within docker (checked manually).

File: tcl/tclcdb2.c [hunk 5]
```diff
@@ -0,0 +1,2379 @@
...
+			   offsetof(cdb2_tm_t, tm_wday)},
+		{  "yday", 0},
+		{    "%d", offsetof(cdb2_client_datetime_t, tm) +
+			   offsetof(cdb2_tm_t, tm_yday)},
+		{ "isdst", 0},
+		{    "%d", offsetof(cdb2_client_datetime_t, tm) +
+			   offsetof(cdb2_tm_t, tm_isdst)},
+		{  "msec", 0},
+		{    "%u", offsetof(cdb2_client_datetime_t, msec)},
+		{"tzname", 0},
```
Comment (akshatsikarwar) [inline tcl/tclcdb2.c]: tzname is optional

File: tcl/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,26 @@
+add_library(tclcdb2 SHARED
+  tclcdb2.c
+)
+
+set(module tclcdb2)
+set(MODULE TCLCDB2)
+configure_file(${PROJECT_SOURCE_DIR}/mem/mem.h.in mem_tcl.h @ONLY)
```
Comment (akshatsikarwar) [inline tcl/CMakeLists.txt]: Don't think we need this.

**Comment by akshatsikarwar** [comment]
LGTM

**Pull request merged**